### PR TITLE
[GEOS-9013] Delegate filter execution matching multiple same-xpath feature chained attribute names to database: Integration test

### PR DESCRIPTION
**Pull request by fernandor777**
Currently app-schema plugin detect two posible attributes with same xpath and delegate filter execution to JVM runtime. Fix delegates filter execution to database using all available join-fields.
This PR adds integration test.

Depends on:
https://github.com/geotools/geotools/pull/2173

Ticket:
https://osgeo-org.atlassian.net/browse/GEOS-9013

### CODE REVIEW

**Comment by nmco** [comment]
As being in stand by for too long, closing it.

File: src/extension/app-schema/app-schema-test/src/test/java/org/geoserver/test/SameNestedAttributeTest.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
+/* (c) 2018 Open Source Geospatial Foundation - all rights reserved
```
Comment (nmco) [inline src/extension/app-schema/app-schema-test/src/test/java/org/geoserver/test/SameNestedAttributeTest.java]: This shoudl be 2019.

**Comment by nmco** [comment]
Looks good to me, since this touch SQL generation code, where both PostGIS and Oracle tests run?

File: src/extension/app-schema/app-schema-test/src/test/java/org/geoserver/test/SameNestedAttributeTest.java (hunk 1 above)
Comment (fernandor777) [reply src/extension/app-schema/app-schema-test/src/test/java/org/geoserver/test/SameNestedAttributeTest.java]: Fixed, thanks.

**Comment by fernandor777** [comment]
Yes, both databases tested.

**Comment by nmco** [comment]
This PR only contributes integration tests merging it.

**Pull request merged**